pkg/cmd/target.go

```diff
@@ -739,6 +769,70 @@ func gardenWrapper(targetReader TargetReader, targetWriter TargetWriter, configR
 	return nil
 }
 
+func serverWrapper(reader ConfigReader, serverName string) {
+	config := reader.ReadConfig(pathGardenConfig)
+
+	if findIP(serverName) != "" {
+		for _, garden := range config.GardenClusters {
+			kc := garden.KubeConfig
+			svr := getServerValueFromKubeconfig(kc)
+			addrIP, err := net.LookupIP(removeHTTPSandPort(svr))
+			checkError(err)
+			if addrIP[0].String() == removeHTTPSandPort(serverName) {
+				pgarden = garden.Name
+			}
+		}
+	} else {
+		for _, garden := range config.GardenClusters {
+			kc := garden.KubeConfig
+			svr := getServerValueFromKubeconfig(kc)
+			if svr == serverName {
+				pgarden = garden.Name
+			}
+		}
+	}
+
+	if pgarden == "" {
+		fmt.Println("the server address enter could not be found as valid Garden target")
+		os.Exit(2)
+	}
+
+}
+
+func findIP(input string) string {
```

Comment:
instead, use https://golang.org/pkg/net/http/cookiejar/?m=all#isIP

Reply:
a bit wired in using the function isIP()
i see it in net/http/cookie package so i did:
import "net/http/cookie"
...
isIP(string) ->didn't pass compile, maybe i used wrong package?

Anyway i checked isIP implementation, the behind logic is `net.ParseIP(str)!=nil`

I tried `net.ParseIP(str)!=nil` to use as if, but in net.ParseIP function, it only returns not nil when the `str` is pure IP address like `127.0.0.1`, it will return nil if the str is `https://127.0.0.1` or `https://127.0.0.1:443`

for my scenario the usage might be https://hostname:443 or https://ipaddress:443, that's why i wrote an function for this....

welcome for suggestion

Reply:
sorry, you are right you can't use what I suggested initially. Meanwhile I updated my comment and the suggestion is included in https://github.com/neo-liang-sap/gardenctl/pull/1